src/Tests/Tests.MonoTorrent.IntegrationTests/IntegrationTests.cs

```diff
@@ -266,15 +270,16 @@ private void OnHttpContext (IAsyncResult ar)
                     }
                     var buffer = new byte[end - start + 1];
                     fs.Seek (start, SeekOrigin.Begin);
-                    fs.Read (buffer, 0, buffer.Length);
-                    ctx.Response.OutputStream.Write (buffer, 0, buffer.Length);
-                    ctx.Response.OutputStream.Close ();
-                    return;
+                    if (fs.Read (buffer, 0, buffer.Length) == buffer.Length) {
+                        ctx.Response.OutputStream.Write (buffer, 0, buffer.Length);
+                        ctx.Response.OutputStream.Close ();
+                    } else {
+                        ctx.Response.StatusCode = 405;
+                        ctx.Response.Close ();
+                    }
```

Comment:
I added a few extra checks here in case fs.Read doesn't read the right number of bytes, or `file == null` etc. I don't think the tests actually hit these issues anyway :P